scipy/sparse/_csparsetools.pyx.in

```diff
@@ -176,7 +176,11 @@ def _lil_get_lengths_{{NAME}}(object[:] input,
 
 {{define_dispatch_map('_LIL_GET_LENGTHS_DISPATCH', '_lil_get_lengths', IDX_TYPES)}}
 
-def lil_flatten_to_array(object[:] input,
+ctypedef fused obj_fused:
```

Comment:
This is unfortunate, but based on https://github.com/cython/cython/issues/2485 it seems like the best workaround for now.

Let's add a comment mentioning the Cython limitation and link to that issue, so that when/if it gets resolved we can remove the hack.